### ISSUE TITLE
Stopped working on Fedora 29 recently?

### ISSUE THREAD

**Issue by dnso86** [mmtrt/notepad-plus-plus#18]
Hi there,

`notepad-plus-plus` stopped working on Fedora 29 the other day. Looking at the timestamps I can see, perhaps it was the update to 7.6.4. that broke it?

```
[@localhost ~]$ notepad-plus-plus 
execl failed: No such file or directory
child exited with status 1
```

```
[@localhost ~]$ snap interfaces | grep notepad-plus-plus
:desktop                         notepad-plus-plus
:desktop-legacy                  notepad-plus-plus
:home                            notepad-plus-plus
:network                         notepad-plus-plus
:opengl                          notepad-plus-plus
:wayland                         notepad-plus-plus
:x11                             notepad-plus-plus
gtk-common-themes:icon-themes    notepad-plus-plus
wine-platform:wine-base-stable   notepad-plus-plus:wine-platform-plug
-                                notepad-plus-plus:cups-control
-                                notepad-plus-plus:hardware-observe
-                                notepad-plus-plus:process-control
-                                notepad-plus-plus:removable-media
```

```
[@localhost ~]$ snap interfaces wine-platform
Slot                             Plug
wine-platform:wine-base-devel    -
wine-platform:wine-base-stable   notepad-plus-plus:wine-platform-plug
wine-platform:wine-base-staging  -
```

```
[@localhost ~]$ sudo snap info notepad-plus-plus
name:      notepad-plus-plus
summary:   Notepad-Plus-Plus is a free source code editor.
publisher: Taqi Raza (mmtrt)
contact:   https://github.com/mmtrt/notepad-plus-plus/wiki
license:   unset
description: |
  Based on the powerful editing component Scintilla, notepad-plus-plus is written in C++ and uses
  pure Win32 API and STL which ensures a higher execution speed and smaller program size. By
  optimizing as many routines as possible without losing user friendliness, notepad-plus-plus is
  trying to reduce the world carbon dioxide emissions. When using less CPU power, the PC can
  throttle down and reduce power consumption, resulting in a greener environment.
commands:
  - notepad-plus-plus
snap-id:      155ZvKiCILXt4S5Ro8YZmzKHgNPPedVj
tracking:     stable
refresh-date: today at 15:28 CET
channels:
  stable:    7.6.4 2019-03-13 (195) 3MB -
  candidate: ↑                          
  beta:      ↑                          
  edge:      7.6.4 2019-03-12 (195) 3MB -
installed:   7.6.4            (195) 3MB -
```

```
[@localhost ~]$ snap version
snap    2.37.4-2.fc29
snapd   2.37.4-2.fc29
series  16
fedora  29
kernel  4.20.16-200.fc29.x86_64
```

**Comment by mmtrt** [comment]
error seems like snapd bug can you try workaround posted in the link report.
https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1756793

**Comment by dnso86** [comment]
Thank you @mmtrt! This bug in fact looks related, however the fix proposed over there (`snap refresh core --beta`) did not solve the problem. I also restarted `snapd` and reinstalled the Notepad++ snap.

Looks like it is something `snap`-related then - I'll give it another go later and keep you updated.

**Comment by mmtrt** [comment]
same issue related to fedora 29 with many other snaps.
https://forum.snapcraft.io/t/compatibility-issues-running-snaps-under-fedora-29/10402

**Comment by dnso86** [comment]
Weird - I still couldn't make it run, even with downgrading `snapd`. Closing the issue though - since likely it has nothing to do with my beloved `notepad-plus-plus`. Thanks for assisting!

**Comment by dnso86** [comment]
FYI - it still persists... Am I the only one?

**Comment by mmtrt** [comment]
can you tell info of snap version

**Comment by dnso86** [comment]
> can you tell info of snap version

```
snap    2.37.4-2.fc29
snapd   2.37.4-2.fc29
series  16
fedora  29
kernel  5.0.5-200.fc29.x86_64
```

**Comment by mmtrt** [comment]
what snap info core says

**Comment by dnso86** [comment]
> what snap info core says

```
name:      core
summary:   snapd runtime environment
publisher: Canonical✓
contact:   snaps@canonical.com
license:   unset
description: |
  The core runtime environment for snapd
type:         core
snap-id:      99T7MUlRhtI3U0QFgl5mXXESAiSwt776
tracking:     beta
refresh-date: 16 days ago, at 20:39 EET
channels:
  stable:    16-2.38                 2019-03-28 (6673) 93MB -
  candidate: 16-2.38                 2019-03-22 (6673) 93MB -
  beta:      16-2.38                 2019-03-21 (6673) 93MB -
  edge:      16-2.38+git1228.a26e409 2019-04-08 (6784) 92MB -
installed:   16-2.38                            (6673) 93MB core
```

I've tried the `edge` channel too now. doesn't work :disappointed:

**Comment by mmtrt** [comment]
same error execl failed?

**Comment by dnso86** [comment]
> same error execl failed? maybe try rebooting system when updated core from edge, also try removing wine snaps i.e notepad-plus-plus and wine-platform.
> 
> I'll setup f29 vm to test this issue.

Same same... did everything you mentioned, then:

```
wine-platform 3.0.4-4.0 from Taqi Raza (mmtrt) installed
notepad-plus-plus 7.6.6 from Taqi Raza (mmtrt) installed
[@localhost ~]$ notepad-plus-plus 
execl failed: No such file or directory
child exited with status 1
```
Thank you for looking into it - I am curious if it is general to Fedora 29 or it is something with my setup...

**Comment by mmtrt** [comment]
It is the issue with snapd on fedora 29 and nothing wrong with your setup. I've tested f29 vm had same issue so then I reported the issue. Snapd devs have acknowledged issue and posted bug report. https://bugs.launchpad.net/snapd/+bug/1824158

**Comment by dnso86** [comment]
> It is the issue with snapd on fedora 29 and nothing wrong with your setup. I've tested f29 vm had same issue so then I reported the issue. Snapd devs have acknowledged issue and posted bug report. https://bugs.launchpad.net/snapd/+bug/1824158

I see - thank you for being so keen on finding an answer. I'll be looking forward for the `snapd` update :)

**Comment by dnso86** [comment]
Finally fixed in `snapd-2.39-1.fc30.x86_64`
